### PR TITLE
Ensure that OmniSharp is launched on global mono even when a version or 'latest' specifed

### PR DESCRIPTION
**Pull request by DustinCampbell**
While @rchande and I were investigating a different issue, we noticed that OmniSharp was always being launched with the run script when `"omnisharp.path"` is set to `"latest"`. Essentially, it would only try to launch OmniSharp on the globally-installed Mono if the user hasn't set `"omnisharp.path"` at all, which really isn't what we want.

Here are the desired behaviors for Linux/OSX:

1. If the user hasn't set `"omnisharp.path"` at all, launch the default OmniSharp. If a valid version of Mono is installed, launch with Mono; otherwise, use the run script.
2. If the user has set `"omnisharp.path"` to an absolute path, assume that path is OmniSharp and launch it. If the user also sets `"omnisharp.useMono"` to `true`, launch with Mono. If a valid version of Mono can't be found, throw an error.
3. If the user has set `"omnisharp.path"` to a version or `"latest"`, first, ensure the appropriate version of OmniSharp is downloaded. Then, if a valid version of Mono is installed, launch with Mono; otherwise, use the run script.

This change fixes `#3` above.

As part of this change, I've done a bit of refactoring to simplify things a bit. The OmniSharpManager has been changed to return two paths: the path to the run script, and the path to launch with Mono. In addition, the calculation of the path to the default OmniSharp has been pushed into the OmniSharpManager. The launcher code has also been simplified a bit. Finally, I changed the logging to only say "started with Mono" if OmniSharp was actually launched with Mono, and I added the version number of Mono.

### CODE REVIEW

File: src/omnisharp/server.ts [hunk 1]
```diff
@@ -312,25 +312,23 @@ export class OmniSharpServer {
...
+            this.eventStream.post(new ObservableEvents.OmnisharpFailure(`Error occured in loading omnisharp from omnisharp.path\nCould not start the server due to ${error.toString()}`, error));
+            return;
         }
 
         this.eventStream.post(new ObservableEvents.OmnisharpInitialisation(new Date(), solutionPath));
         this._fireEvent(Events.BeforeServerStart, solutionPath);
 
-        return launchOmniSharp(cwd, args, launchPath).then(async value => {
-            this.eventStream.post(new ObservableEvents.OmnisharpLaunch(value.usingMono, value.command, value.process.pid));
+        return launchOmniSharp(cwd, args, launchInfo).then(async launchResult => {
```
Comment (akshita31) [inline src/omnisharp/server.ts]: Can we use async await here as well ?

Comment (DustinCampbell) [reply src/omnisharp/server.ts]: Sure

Comment (akshita31) [reply src/omnisharp/server.ts]: Thanks!

**Comment by DustinCampbell** [comment]
Hmmm... I"m not sure what's up with codecov. It seems to show lots of uncovered blocks in OmniSharpManager.ts, but those blocks are clearly covered by feature tests. @TheRealPiotrP, is this a known issue?

**Comment by akshita31** [comment]
@DustinCampbell  We do not have code coverage for feature tests right now. I have created an issue and will look into it - https://github.com/OmniSharp/omnisharp-vscode/issues/2227

**Comment by DustinCampbell** [comment]
Thanks @akshita31! Are you all ok if I go ahead and merge this in that case? The OmniSharpManager.ts file is really quite well-covered if feature tests are included.

**Comment by akshita31** [comment]
@DustinCampbell Sure go ahead 😀

**Pull request merged**